### DISCUSSION TITLE
Get wikijs version as a low-privilege user

### DISCUSSION THREAD

**Discussion by AndrewSav** [requarks/wiki#6463] [Help / Questions] [answered]
This gets me the wikijs version: `curl 'https://wikjs/graphql' -H 'authorization: Bearer token' -H 'content-type: application/json' --data-raw '{"query":"{system{info{currentVersion latestVersion}}}"}'`. However the token seems to require to have `manage:system` for this to work, which feels like a huge overkill. How can I fetch the version(s) in a secure manner?

**Comment by NGPixel** [comment]
This is by design as advertising the version of any server software is a potential security risk.

**Reply by AndrewSav** [reply]
@NGPixel  I'm sorry, can you explain this a bit more. Imagine that all an application need to do with wikijs is to query its version. This is not a hypothetical, I have an application like that. It does not need to be able to do anything else, and it does not need to be able to perform potentially security risky functions.

Can you explain, how giving that application the admin token is more secure, than giving it a limited token that allows it only to do a limited functionality that it needs to perform (querying the version)? This is very unintuitive to me so I would love to see an expanded explanation, thank you in advance! Intuitively it is less secure, not more, because if the token leaks, an attacked could do all sorts of potentially dangerous actions, they could not do with a non `manage:system` token.

**Reply by AndrewSav** [reply]
The way I see it, and please do correct me if I'm wrong, a good practice is to follow the [Principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege). This means, that we give a token only those privileges which are essential to perform its intended function. In this case it's fetching the wikijs version. I said that using `manage:system` is a huge overkill, because it seems to be the opposite to the Principle of least privilege: we give all possible rights including admin rights to a entity that does not need them, does not need any write or manage access, and only need to fetch the version. I said that I'd like to fetch the version(s) in a secure manner, because I perceived the above as a security risk, so I was hoping for a scope that has less permissions than `manage:system` to fetch the version(s).

I do understand that a developer can be granular only so much. It might not be practical to give a scope to each and every possible action. In this particular case a definite improvement would be a scope along the lines of `read:system` that would provide the read access to the system information without being able to actually change anything. Of course, implementing this one can go as granular as one likes, and have a specific scope to read the version(s), if deemed practical.

Now, I was not able to find a suitable scope here, that I can give the application to read the version(s) but to be unable to do much damage in the worst case, so I thought I might have missed something.

Any more explanation would be greatly appreciated. Apologies, for being verbose, just want to make sure that I explained my query the best I can.

**Reply by AndrewSav** [reply]
I updated the title from "Get wikijs version as an unprivileged user" to "Get wikijs version as a low-privilege user", to emphasise the fact that I did not ask for a facility to get the version without a token, and asked "how can we get a version with giving the admin rights?" instead, I think there might have been misunderstanding.

**Reply by NGPixel** [reply, marked as answer]
My answer was strictly regarding the version being made public (e.g. without a token / guest), which is a bad idea.

Making it available via a specific permission could be done but is not possible at the moment. The GraphQL API was made to serve the Wiki.js UI, where the version is displayed only to admins.

The upcoming v3 will have more granular permissions to allow use cases outside of Wiki.js itself.
